### PR TITLE
feat(protocol): Deposit Tx Utility

### PR DESCRIPTION
**Pull request by refcell**
### Description

Adds a simple utility method to tell if a tx is a deposit transaction.

### CODE REVIEW

File: crates/protocol/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//! Utility methods used by protocol types.
+
+/// Returns if the given `value` is a deposit transaction.
+pub fn is_deposit<B>(value: &B) -> bool
+where
+    B: AsRef<[u8]>,
+{
+    !value.as_ref().is_empty() && value.as_ref()[0] == 0x7E
```
Comment (mattsse) [inline crates/protocol/src/utils.rs:8]:
```suggestion
    value.as_ref().first() == Some(&0x7E)
```

File: crates/protocol/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+//! Utility methods used by protocol types.
+
+/// Returns if the given `value` is a deposit transaction.
+pub fn is_deposit<B>(value: &B) -> bool
```
Comment (mattsse) [inline crates/protocol/src/utils.rs:4]: perhaps rename this to `starts_with_2781_deposit` or smth?

**Review by mattsse — CHANGES_REQUESTED** [review]
also fyi:

https://github.com/alloy-rs/alloy/blob/fb7b09ed92c5f938a36803e9fda688c8a99bcbe2/crates/eips/src/eip2718.rs#L64-L69

File: crates/protocol/src/utils.rs (hunk 2 above)
Comment (refcell) [reply crates/protocol/src/utils.rs:4]: Ah nice, will do

**Review by mattsse — APPROVED** [review]

**Pull request merged**